python-package/securexgboost/core.py

```diff
@@ -2040,3 +2299,309 @@ def get_split_value_histogram(self, key, feature, fmap='', bins=None, as_pandas=
+        filenames = list(request.filenames)
+        usernames = list(request.usernames)
+        silent = request.silent
+
+        dmat_handle = ctypes.c_char_p()
+        _check_call(_LIB.XGDMatrixCreateFromEncryptedFile(
+            from_pystr_to_cstr(filenames),
+            from_pystr_to_cstr(usernames),
+            c_bst_ulong(len(filenames)),
+            ctypes.c_int(silent),
+            ctypes.byref(dmat_handle)))
+        return dmat_handle.value.decode('utf-8')
+
+
+    def XGBoosterSaveModel(request):
+        booster_handle = request.booster_handle
+        filename = request.filename
+        username = request.username
+
+        _check_call(_LIB.XGBoosterSaveModel(
+            c_str(booster_handle),
+            c_str(filename),
+            c_str(username)))
+
+    def XGBoosterLoadModel(request):
+        booster_handle = request.booster_handle
+        filename = request.filename
+        username = request.username
+
+        _check_call(_LIB.XGBoosterLoadModel(
+            c_str(booster_handle),
+            c_str(filename),
+            c_str(username)))
+
+    # TODO test this
+    def XGBoosterDumpModelEx(request):
+        booster_handle = request.booster_handle
+        fmap = request.fmap
+        with_stats = request.with_stats
+        dump_format = request.dump_format
+
+        length = c_bst_ulong()
+        sarr = ctypes.POINTER(ctypes.c_char_p)()
+        _check_call(_LIB.XGBoosterDumpModelEx(
+            c_str(booster_handle),
+            c_str(fmap),
+            ctypes.c_int(with_stats),
+            c_str(dump_format),
+            ctypes.byref(length),
+            ctypes.byref(sarr)))
+        return length.value, from_cstr_to_pystr(sarr, length)
+
+    def XGBoosterDumpModelExWithFeatures(request):
+        booster_handle = request.booster_handle
+        flen = request.flen
+        fname = request.fname
+        ftype = request.ftype
+        with_stats = request.with_stats
+        dump_format = request.dump_format
+
+        length = c_bst_ulong()
+        sarr = ctypes.POINTER(ctypes.c_char_p)()
+        _check_call(_LIB.XGBoosterDumpModelExWithFeatures(
+            c_str(booster_handle),
+            ctypes.c_int(flen),
+            from_pystr_to_cstr(list(fname)),
+            from_pystr_to_cstr(list(ftype)),
+            ctypes.c_int(with_stats),
+            c_str(dump_format),
+            ctypes.byref(length),
+            ctypes.byref(sarr)))
+        return length.value, from_cstr_to_pystr(sarr, length)
+
+    # TODO test this
+    def XGBoosterGetModelRaw(request):
+        booster_handle = request.booster_handle
+        username = request.username
+
+        length = c_bst_ulong()
+        sarr = ctypes.POINTER(ctypes.c_char_p)()
+        _check_call(_LIB.XGBoosterGetModelRaw(
+            c_str(booster_handle),
+            c_str(username)))
+        return length.value, from_cstr_to_pystr(sarr, length)
+
+    def XGDMatrixNumCol(request):
```

Comment:
If we don't expose `num_row()` and `num_col()` in the DMatrix API, I think we can comment these two out.

Reply:
See above.